### ISSUE TITLE
BatchNorm backprop fails at test time

### ISSUE THREAD

**Issue by Fredrik-M** [deeplearning4j/deeplearning4j#9541]
#### Issue Description
Backpropagation crashes in BatchNorm layer when the forward pass was executed at test time. This is due to a shape mismatch:
```
Caused by: java.lang.IllegalStateException: Cannot perform in-place operation "muli": result array shape does not match the broadcast operation output shape: [2].muli([1, 2]) != [2].
In-place operations like x.muli(y) can only be performed when x and y have the same shape, or x and y are broadcastable with x.shape() == broadcastShape(x,y)
	at org.nd4j.linalg.api.shape.Shape.assertBroadcastable(Shape.java:281)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.muli(BaseNDArray.java:3237)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.muli(BaseNDArray.java:3231)
	at org.deeplearning4j.nn.layers.normalization.BatchNormalization.backpropGradient(BatchNormalization.java:294)
	at org.deeplearning4j.nn.graph.vertex.impl.LayerVertex.doBackward(LayerVertex.java:148)
	at org.deeplearning4j.nn.graph.ComputationGraph.calcBackpropGradients(ComputationGraph.java:2772)
```
In my case the layer input is convolution feature maps, so the problem is https://github.com/eclipse/deeplearning4j/blob/7da2fd5ea27662dd1221eacf7bb14bb56a759fae/deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/normalization/BatchNormalization.java#L294
`dxhat` has shape `[N,C,H,W]` in this case, so summing over non channel dimensions yields shape `[C]`. At training time, the mean and variance are computed by `x.mean` and `x.var`, which in this case return arrays of shape `[C]`, hence the field `std` in BatchNormalization class also has this shape at training time.
However, the parameters for global mean and variance estimates have shape `[1,C]`, so when running backprop at test time, `std` will not match the shape of `dxhat.mul(xMu).sum(nonChDims).muli(-0.5)` (we get `[C].muli([1,C])`).

#### Version Information

```
<dependency>
      <groupId>org.deeplearning4j</groupId>
      <artifactId>deeplearning4j-core</artifactId>
      <version>1.0.0-M1.1</version>
</dependency>
<dependency>
      <groupId>org.nd4j</groupId>
      <artifactId>nd4j-native-platform</artifactId>
      <version>1.0.0-M1.1</version>
</dependency>
```

**Comment by agibsonccc** [comment]
@Fredrik-M mind getting me a reproducer class for this? This might already be fixed on master. Thanks!

**Comment by agibsonccc** [comment]
Closing due to inactivity
